crates/bevy_ecs/src/system/query.rs

```diff
@@ -267,6 +267,20 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
         }
     }
 
+    /// Convert the query to readonly.
+    pub fn to_readonly(&self) -> Query<'w, 's, Q::ReadOnly, F::ReadOnly> {
```

Comment:
this is unsound and will let you have a `Query<&mut T>` and `Query<&T>` at the same time, the type signature needs to be 
```suggestion
    pub fn to_readonly(&self) -> Query<'_, '_, Q::ReadOnly, F::ReadOnly> {
```
can you add a `bevy_ecs_compile_fail_tests` test that after calling `to_readonly` any attempt to access the original query mutably (or by value) will invalidate the readonly query

Reply:
Okay note taken.